aws/rust-runtime/aws-inlineable/src/http_request_checksum.rs

```diff
@@ -130,6 +131,22 @@ where
         cfg: &mut ConfigBag,
     ) -> Result<(), BoxError> {
         let checksum_algorithm = (self.algorithm_provider)(context.input())?;
+        if let Some(checksum_algorithm) = checksum_algorithm {
```

Comment:
Would probably remove these changes for now and just roll out this detection with the new flex checksums (mostly in favor of making my merge to the feature branch easier)